### DISCUSSION TITLE
Feed_audio transcription

### DISCUSSION THREAD

**Discussion by Baavro** [KoljaB/RealtimeSTT#17] [Q&A]
I am unable to use feed_audio for transcription. It would be great if you could make a code snippet for the same.
i am sending data as 
const audioBlob = new Blob(audioChunks, { type: 'audio/wav' });

      const reader = new FileReader();
      reader.readAsDataURL(audioBlob);

      reader.onloadend = () => {
        const audioBase64 = reader.result as string;

        const message = JSON.stringify({
          type: 'audio',
          spokenText: spokenText,
          content: audioBase64,
        });
        
        then I am feeding the content to feed_audio.

data_url = data.get("content")
                            _, base64_data = data_url.split(',', 1)
                            #audio_chunk = data.get("content")
                            audio_data = base64.b64decode(base64_data)
                            print("audio_file converted")
                            recorder.feed_audio(audio_data)
                            print("audio_file fed")

But I am unable to transcribe this

**Comment by KoljaB** [comment]
[This example](https://github.com/KoljaB/RealtimeSTT/blob/master/example_browserclient/server.py) showcases the usage.
feed_audio expects chunks to be in raw 16-bit PCM mono format with samplerate 16000. Base64 is just a wrapper, it does not tell us about the internal format of the audio data. You need to ensure that the Base64-decoded data is converted into raw PCM format (if it isn't already) and matches the expected parameters (mono, 16 bit, 16kHz sample rate)  before feeding it into the feed_audio method for transcription.
